bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/NetatmoConstants.java

```diff
@@ -0,0 +1,322 @@
+        public final int maxValue;
+        public final int scale;
+        public final Unit<?> unit;
+
+        private Measure(double minValue, int maxValue, double precision, Unit<?> unit) {
+            this.minValue = minValue;
+            this.maxValue = maxValue;
+            this.unit = unit;
+            String[] splitter = Double.valueOf(precision).toString().split("\\.");
+            if (splitter.length > 1) {
+                int dec = Integer.parseInt(splitter[1]);
+                this.scale = dec > 0 ? Integer.toString(dec).length() : 0;
+            } else {
+                this.scale = 0;
+            }
+        }
+    }
+
+    public static class MeasureChannelDetails {
+        public final URI configURI;
+        public final String itemType;
+        public final String pattern;
+
+        public MeasureChannelDetails(String measureType, String itemType, String pattern) {
+            this.configURI = URI.create(String.join(":", BINDING_ID, measureType, "config"));
+            this.itemType = itemType;
+            this.pattern = pattern;
+        }
+    }
+
+    public enum MeasureClass {
+        INTERIOR_TEMPERATURE(new Measure(0, 50, 0.3, SIUnits.CELSIUS), true, "TEMP", "Temperature", "Temperature"),
+        EXTERIOR_TEMPERATURE(new Measure(-40, 65, 0.3, SIUnits.CELSIUS), true, "TEMP", "Temperature", "Temperature"),
+        PRESSURE(new Measure(260, 1260, 1, HECTO(SIUnits.PASCAL)), true, "PRESSURE", "Pressure", "Pressure"),
+        CO2(new Measure(0, 5000, 50, Units.PARTS_PER_MILLION), true, "CO2", "CO2", "Dimensionless"),
+        NOISE(new Measure(35, 120, 1, Units.DECIBEL), true, "NOISE", "Noise", "Dimensionless"),
+        RAIN_QUANTITY(new Measure(0, 150, 0.1, MILLI(SIUnits.METRE)), false, "SUM_RAIN", "Rain", "Length"),
+        RAIN_INTENSITY(new Measure(0, 150, 0.1, Units.MILLIMETRE_PER_HOUR), false, "", "Rain", "Speed"),
+        WIND_SPEED(new Measure(0, 160, 1.8, SIUnits.KILOMETRE_PER_HOUR), false, "", "Wind", "Speed"),
+        WIND_ANGLE(new Measure(0, 360, 5, Units.DEGREE_ANGLE), false, "", "Wind", "Angle"),
+        HUMIDITY(new Measure(0, 100, 3, Units.PERCENT), true, "HUM", "Humidity", "Dimensionless");
+
+        public static final EnumSet<MeasureClass> asSet = EnumSet.allOf(MeasureClass.class);
+        private static final String MEASURE = "measurement";
+
+        public final Measure measureDefinition;
+        public final String apiDescriptor;
+        public final String tagName;
+        public final Map<String, MeasureChannelDetails> channels = new HashMap<>(2);
+
+        MeasureClass(Measure measureDef, boolean canScale, String apiDescriptor, String tagName, String dimension) {
+            this.measureDefinition = measureDef;
+            this.apiDescriptor = apiDescriptor;
+            this.tagName = tagName;
+            if (!apiDescriptor.isBlank()) {
+                String measureType = MEASURE;
+                if (apiDescriptor.equals("SUM_RAIN")) {
```

Comment:
Suggest constant on left side to prevent NullPointerException in case of refactoring. Perhaps also introduce constant for "SUM_RAIN"?

Reply:
Done